src/modules/rlm_eap/types/rlm_eap_ttls/ttls.c

```diff
@@ -415,8 +415,11 @@ static VALUE_PAIR *diameter2vp(REQUEST *request, REQUEST *fake, SSL *ssl,
 				return NULL;
 			}
 
-			eapttls_gen_challenge(ssl, challenge,
-					      sizeof(challenge));
+			/*
+			 *	TLSv1.3 exports a different key depending on the length
+			 *	requested so ask for *exactly* what the spec requires
+			 */
+			eapttls_gen_challenge(ssl, challenge, vp->vp_length + 1);
```

Comment:
https://tools.ietf.org/html/rfc5281#section-11.2.4 says 16 octets of challenge, followed by 1 octet of `ident`.

We might want to check the `ident`, too...

Reply:
The comment above this (vintage deKok circa 2003): "We're a little forgiving in that we have loose checks on the length, and we do NOT check the Id (first octet of the response to the challenge). But if the client gets the challenge correct, we're not too worried about the Id."

I suspect this is because this code is in the middle of the VP iterator and may not have seen the challenge yet. Do you want me to break the check out to actually look at the ID?